### ISSUE TITLE
expose a method to check if player is Loaded ...

### ISSUE THREAD

**Issue by sallypeters** [erickzanardo/ocarina#3]
Hey Erick, thank you for this plugin. I can confirm so far testing in iOS it works perfectly and I have a gapless playback.

I use Ocarina in a custom AudioClass that has other functionality inbuilt. I would supply an asset to this custom class and call Ocarina methods to play/pause/resume accordingly.

To make better - would be great if :

1- You expose a method to check if the player is already loaded  : 

```
bool isLoaded(){
 return (_id!=null);
}

```

This is required because no pause/play/resume etc method in Ocarina can be called unless the Player has been loaded and we need a way to instantiate and prepare Ocarina player if its used inside another class and the asset/filepath is supplied in the constructor of a custom class. ie.

```

class AudioLoop{
  bool playMusic = false;
  final String assetToPlay;
  final OcarinaPlayer _audioPlayer;
 

  AudioLoop({@required this.assetToPlay})
      : _audioPlayer = OcarinaPlayer(asset: assetToPlay, loop: true) {
    // Created but not loaded and you cannot call _audioPlayer.load() here
  }

  play({bool play: false}) async {
    playMusic = play;
    if (playMusic )
   {
/// Would be better to have :
/// if (!_audioPlayer.isLoaded()){
///await _audioPlayer.load();
///} 
       await _audioPlayer.load(); 
      _audioPlayer.play();
    }
  }
}
```
2- Unless you want to allow load() to be repeatably called, we optimise Load() so it checks if _id is null and only loads inits if this is the case :

```
Future<void> load() async {
    if (_id!=null) return;

_id = await  _channel.invokeMethod('load', {
      'url': asset ?? filePath,
      'volume': volume,
      'loop': loop,
      'isAsset': asset != null,
    });
  }

```  

3- Add a dispose so we can cleanup resources

**Comment by erickzanardo** [comment]
sounds reasonable enough, gonna make it when I get some time.

**Comment by erickzanardo** [comment]
Released on `0.0.2`
